### DISCUSSION TITLE
Generate Token from a Function

### DISCUSSION THREAD

**Discussion by SimbaNinja917** [beholdr/maska#106] [Q&A]
Hello,
I am using a function called `generateToken` inside the `<script setup lang="ts">` to generate a mask token. 

I am getting a `500` `error` on Nuxt, `Expected property name or '}' in JSON at position 1`

Below is the relevant part of the code, what am I doing wrong? 

```<script setup lang="ts">
import {vMaska} from "maska";

function generateToken(word: string) {
// function logic goes here
    return token;
}
</script> 

<template>
<input v-maska data-maska-eager="true" data-maska="{{selectedWord}}" data-maska-tokens="{{generateTokent(selectedWord)}}"> 
</template>
```

Thank you

**Comment by beholdr** [comment]
I'm not sure I understand what result do you want to get? What are selectedWord and logic in generateToken functions? Do you want a dynamic tokens?

**Reply by SimbaNinja917** [reply]
Yes, I want to generate dynamic masks and tokens, for a word-guessing game where the end-user cannot input a letter except the correct one for each letter. 
Desired Result Example:
```
let options = { mask: 'cat', tokens: { c: { pattern: /[Cc]/}, a: { pattern: /[Aa]/}, t: { pattern: /[Tt]/} } }
let options = { mask: 'dog', tokens: { d: { pattern: /[Dd]/}, o: { pattern: /[Oo]/}, g: { pattern: /[Gg]/} } }
```
etc. 

I have come-up with this: 
```
let option = {
mask: word, // cat
tokens: generateInputMaskRegEx(word) // tokens: { c: { pattern: /[Cc]/}, a: { pattern: /[Aa]/}, t: { pattern: /[Tt]/} }
}
```

```
function generateInputMaskRegEx(word: string): Object {           // cat
    let regexResponse={};
    for (const [key, value] of Object.entries(word)) {
        regexResponse[value] = { pattern: [word[key].toUpperCase() + word[key].toLowerCase()]}; // c: { pattern: /[Cc]/}
    } 
    return regexResponse; // tokens: { c: { pattern: /[Cc]/}, a: { pattern: /[Aa]/}, t: { pattern: /[Tt]/} }
}
```

```  <input v-maska:[options] data-maska-eager="true" data-maska:[options]>  ```

What I'm getting: 
```
"tokens": { "c": { "pattern": "/[Cc]/" }, "a": { "pattern": "/[Aa]/" }, "t": { "pattern": "/[Tt]/" } } }
```

and it's not working...

**Reply by SimbaNinja917** [reply]
I've also tried the custom tokens by ```data-attr``` as mentioned in the docs [here](https://beholdr.github.io/maska/#/?id=custom-tokens), which is not working neither: 

```
function generateInputMaskRegEx(word: string): String { // cat
    let regexResponse='';
    for (const [key, value] of Object.entries(word)) { 
        regexResponse = regexResponse.concat(word[key].toLowerCase()+':'+word[key].toUpperCase() + word[key].toLowerCase()+'|'); // c:Cc|
    } 
    regexResponse = regexResponse.slice(0, -1) // we delete the last character's "|" // c:Cc|a:Aa|t:Tt
    return regexResponse; // c:Cc|a:Aa|t:Tt

}
```

```
<input v-maska:generateInputMaskRegEx(word) data-maska-eager="true" data-maska:{{generateInputMaskRegEx(word)}}>  
```

**Reply by beholdr** [reply]
Well, I tried to implement this logic, and it worked. You can’t make data-attributes reactive: your mask and tokens should be reactive (computed) and set via options like this:

``` html
<script setup>
import { computed, ref } from 'vue'
import { vMaska } from 'maska'

const cat = ref(true)

const options = computed(() =>
  cat.value
    ? {
        mask: 'cat',
        tokens: {
          c: { pattern: /[Cc]/ },
          a: { pattern: /[Aa]/ },
          t: { pattern: /[Tt]/ }
        }
      }
    : {
        mask: 'dog',
        tokens: {
          d: { pattern: /[Dd]/ },
          o: { pattern: /[Oo]/ },
          g: { pattern: /[Gg]/ }
        }
      }
)
</script>

<template>
  <label><input type="checkbox" v-model="cat" />Is cat?</label><br />
  <input v-maska:[options] />
</template>
```

[Example in Vue playground](https://sfc.vuejs.org/#eNqFkk1v2zAMhv8KoUtSILWAHQ0nXbEBww7DLrtVPSgyk7qxPiDR3oYg/32U7SWpg6I+keLDlzTJo3gMoeg7FKWokolNIEhIXdgo19jgI8ERjLehI6xXEHEHJ9hFb2HBSYsrqP+h00GfozZ7HFfOeJcIjCZY5/wlxQ7vLgEfqGGDg//LLJd3sOb6kJOKXrcdZgfgAY6jkb9coIQFI4vV5ZX8AV0qr0GWYR+CJsLoSpBPX8yzhNNVFoCeIY/6BqEZ8osyciHO5mS8aWLqtvb7D7utZ3W+1jet+Bny098g+xnybf9+t7yNSo7L57GzQ2hDqwmHJVSt3mK7qRrH2wH6G3CthHlBc9j6P0pAf299jW1+1MS+3HxPeXMPlZwyt5EfB6lRgzPydZRP0+6fh3Alr8qKlRgvi9FQvCbv+ECHMakpkJQ4D04JPsbsK/FCFFIpZdqZfNavqfBxL9kqYueosVhgsvfb6H8njCysxDQ2JYam3qqY2jHDf9f0sXBI0gUrB+7zJ1k3iUZn0MkyPMyTOP0DCxYIbQ==)
